test/e2e_node/image_pull_test.go

```diff
@@ -63,15 +63,15 @@ var _ = SIGDescribe("Pull Image", feature.CriProxy, framework.WithSerial(), func
 			if err := resetCRIProxyInjector(e2eCriProxy); err != nil {
 				ginkgo.Skip("Skip the test since the CRI Proxy is undefined.")
 			}
-
+			ginkgo.DeferCleanup(func() {
+				err := resetCRIProxyInjector(e2eCriProxy)
+				framework.ExpectNoError(err)
+			})
 			testpods = prepareAndCleanup(ctx, f)
 			gomega.Expect(len(testpods)).To(gomega.BeNumerically("<=", 5))
 		})
 
 		ginkgo.AfterEach(func(ctx context.Context) {
-			err := resetCRIProxyInjector(e2eCriProxy)
-			framework.ExpectNoError(err)
-
```

Comment:
No need to put it into this PR, but this entire AfterEach could get replaced by a `DeferCleanup` after creating each test pod. The `testpods` variable then isn't needed anymore.

Reply:
Thanks, good point, code will be cleaner. I am not sure if testpods isn't  needed anymore, seems is used internally in other places as well.  Please take another look @pohly

According to [Ginkgo docs](https://onsi.github.io/ginkgo/#suite-setup-and-cleanup-beforesuite-and-aftersuite) , 

> ... You can also pass a function that returns values. DeferCleanup ignores all these return value except for the last. If the last return value is a non-nil error - a common go pattern - DeferCleanup will fail the spec...:

So cleaned up  the code a bit more, what do you think ?

https://github.com/kubernetes/kubernetes/blob/9e1ebafcb5ad3aa42bd055e22c7a3b82ee68abec/test/e2e_node/container_restart_test.go#L56-L58